skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillsController.java

```diff
@@ -0,0 +1,69 @@
+package com.RDS.skilltree.Skill;
+
+import com.RDS.skilltree.Exceptions.NoEntityException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.coyote.Response;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.ObjectUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+@RestController
+@Slf4j
+@RequestMapping("/v1/skills")
+public class SkillsController {
+    private final SkillsService skillsService;
+    public SkillsController(SkillsService skillsService){
+        this.skillsService = skillsService;
+    }
+
+    @PostMapping("/")
+    public ResponseEntity<?> createSkill(@RequestBody(required = true) SkillDRO skillDRO){
+        if (skillDRO.getCreatedBy() == null || skillDRO.getType() == null || skillDRO.getName() == null){
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body("CreatedBy, Type and Name are mandatory values, anyone cannot be null");
+        }
+        try {
+            return ResponseEntity.status(HttpStatus.CREATED).body(skillsService.createSkill(skillDRO));
+        } catch(DataIntegrityViolationException ex){
+            return ResponseEntity.status(HttpStatus.CONFLICT)
+                    .body("Cannot create entry for Skill as Skill name is duplicate");
+        }
```

Comment:
What if there is a different type of exception? Eg: NullPointer exception

Reply:
What if any of the exceptions listed in this superset occurs? 
<img width="577" alt="Screenshot 2023-12-07 at 10 44 33 PM" src="https://github.com/Real-Dev-Squad/skill-tree-backend/assets/98796547/e4d91bc1-5a36-4f98-8c0f-1b5654f67044">

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/dao/DataIntegrityViolationException.html

Reply:
The end user was able to see the internal server error, will once need your help to debug this and change it to the exception message we are throwing, not just Internal server error @Ajeyakrishna-k